src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -160,6 +165,8 @@ SGVector<float64_t> CGaussianProcessMachine::get_posterior_variances(
 
 	// get kernel matrix and create eigen representation of it
 	SGMatrix<float64_t> k_tsts=kernel->get_kernel_matrix();
+	//To DO in order to speed up the prediction process,
+	//we only need to obtain the diagonal elements of the kernel matrix
```

Comment:
@lambday  
`get_parameter_gradient` is a method in `CKernel`

I may add a method called `get_parameter_gradient_diagonal` in `CKernel`, which will naively call get_parameter_gradient(param, i).get_diagonal_vector()

In ARD kernels, It seems  `get_parameter_gradient_diagonal` can be efficiently implemented.